openraft/src/engine/engine_impl.rs

```diff
@@ -1136,11 +1136,10 @@ where
             return Err(RejectVoteRequest::ByVote(self.state.vote));
         }
 
-        tracing::debug!(%vote, "vote is changing to" );
-
         // Grant the vote
 
         if vote > &self.state.vote {
+            tracing::debug!(%vote, "vote is changing to" );
```

Comment:
Please keep the `debug!` before this condition check: for debugging updating the `vote` with a same value may be an unexpected behavior:)